external/llvm-project/mlir/include/mlir/Interfaces/CallInterfaces.td

```diff
@@ -32,6 +32,24 @@ def CallOpInterface : OpInterface<"CallOpInterface"> {
   let cppNamespace = "::mlir";
 
   let methods = [
+    InterfaceMethod<
+      /*desc=*/[{
+        Clone the current call-like operation with the given attributes but
+        re-assign operands and result-types. Used in bufferization for result
+        buffers.
+      }],
+      /*retTy=*/"Operation *",
+      /*methodName=*/"clone",
+      (ins "OpBuilder &":$b, "Location":$loc, "TypeRange":$resultTypes,
+           "ValueRange":$operands),
+      [{
+        BlockAndValueMapping bvm;
+        OperationState state(
+          loc, ConcreteOp::getOperationName(), operands, resultTypes,
+          $_op->getAttrs());
+        return b.create(state);
+      }]
+    >,
```

Comment:
Why's this being re-added (is it being re-added?)? Does its upstream replacement, if any, non work?

Reply:
This is no upstream clone for CallOpInterface's. 

Bufferization was only written to work for func::FuncOp, but this PR enhances to support async.launch ops.

Reply:
Right, I now see why the generic clone() doesn't work.